cmd/kube-proxy/app/server.go

```diff
@@ -204,14 +207,31 @@ func (o *Options) Complete() error {
 		}
 	}
 
-	err := utilfeature.DefaultFeatureGate.SetFromMap(o.config.FeatureGates)
-	if err != nil {
+	if err := o.processHostnameOverrideFlag(); err != nil {
+		return err
+	}
+
+	if err := utilfeature.DefaultFeatureGate.SetFromMap(o.config.FeatureGates); err != nil {
 		return err
 	}
 
 	return nil
 }
 
+// processHostnameOverrideFlag processes hostname-override flag
+func (o *Options) processHostnameOverrideFlag() error {
+	// Check if hostname-override flag is set and use value since configFile always overrides
```

Comment:
I thought the flag would always override - the flag reads into the config struct..

@mtaufen

Reply:
@thockin I did kube-proxy first, as a POC, and it got merged well before all the docs/discussions on the proper way to do things going forward. I didn't realize that there would be a need for per-instance configuration settings as well as those that are shared across all instances. The way kube-proxy specifically works, today, is that it's completely either-or. Either you use a config file, or you use flags. Flags do not override the config file.

Reply:
blech